src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ParseJson.txt

```diff
@@ -87,6 +87,54 @@ Blank()
 >> Sum(Table(ParseJSON("[1, 2, 3, 4, 5]")), Value(Value))
 15
 
+>> ForAll(ParseJSON("[5, 6, 7]"), Value(ThisItem))
+[5,6,7]
+
+>> ForAll(ParseJSON("null"), Value(ThisItem))
+Blank()
+
+>> ForAll(ParseJSON("[null, null, null]"), Value(ThisItem))
+[Blank(),Blank(),Blank()]
+
+>> ForAll(ParseJSON("[5, null, 7]"), Value(ThisItem))
+[5,Blank(),7]
+
+>> ForAll(ParseJSON("[""hi"", null, ""hello""]"), Text(ThisItem))
+["hi",Blank(),"hello"]
+
+>> ForAll(ParseJSON("[true, null, false]"), Boolean(ThisItem))
+[true,Blank(),false]
+
+>> ForAll(ParseJSON("[""2011-01-15"", null]"), Text(DateValue(ThisItem), "yyyy-MM-dd"))
+["2011-01-15",""]
+
+>> ForAll(ParseJSON("[""08:03:05.000"", null]"), Text(TimeValue(ThisItem), "HH:mm:ss"))
+["08:03:05",""]
+
+>> ForAll(ParseJSON("[""5cc45615-f759-4a53-b225-d3a2497f60ad"", null]"), GUID(ThisItem))
+[GUID("5cc45615-f759-4a53-b225-d3a2497f60ad"),Blank()]
+
+>> ForAll(ParseJSON("[1]"), Value(ThisItem) / 0)
+#Error
+
+>> CountRows(ForAll(ParseJSON("[1,2,3]"), Value(ThisItem)))
+3
+
+>> Sum(ForAll(ParseJSON("[5, 6, 7]"), Value(ThisItem)), Value)
+18
+
+>> Sum(ForAll(ParseJSON("[5, 6, 7]"), ThisItem), Value(Value))
+18
+
+>> ForAll(ParseJSON("[1,{""a"":false},""hello""]"), IfError(Text(ThisItem), IfError(Text(Value(ThisItem)), Text(Boolean(ThisItem.a)))))
+["1","false","hello"]
+
+>> ForAll(ParseJSON("[1,true,3,false]"), Value(ThisItem))
```

Comment:
>ForAll(ParseJSON("[1,true,3,false]"), Value(ThisItem)) [](http://example.com/codeflow?start=3&length=54)

Would following be an error?

ForAll(ParseJSON("[1,2,3]"), ThisRecord)

Reply:
Yes, ThisRecord symbol not found

Reply:
Added this test